draft-ietf-taps-arch.md

```diff
@@ -325,90 +335,90 @@ The diagram below provides a high-level view of the actions taken during the lif
+* Connection Finished: Signals to an application that a given Connection is no longer usable for sending or receiving Messages. The event SHOULD deliver an error to the application that describes the nature of the termination.
 
 * Connection Received: Signals to an application that a given Listener has passively received a Connection.
 
-* Message Received: Delivers received Message content to the application, based on a Receive action. This may include an error if the Receive action cannot be satisfied due to the Connection being closed.
+* Message Received: Delivers received Message content to the application, based on a Receive action. This MAY include an error if the Receive action cannot be satisfied due to the Connection being closed.
 
-* Message Sent: Notifies the application of the status of its Send action. This may be an error if the Message cannot be sent, or an indication that Message has been processed by the protocol stack.
+* Message Sent: Notifies the application of the status of its Send action. This might indicate a failure if the Message cannot be sent, or an indication that Message has been processed by the protocol stack.
 
-* Path Properties Changed: Notifies the application that some property of the Connection has changed that may influence how and where data is sent and/or received.
+* Path Properties Changed: Notifies the application that some property of the Connection has changed that might influence how and where data is sent and/or received.
 
 ### Termination Actions {#termination}
 
-* Close: The action an application may take on a Connection to indicate that it no longer intends to send data, is no longer willing to receive data, and that the protocol should signal this state to the remote endpoint if applicable.
+* Close: The action an application takes on a Connection to indicate that it no longer intends to send data, is no longer willing to receive data, and that the protocol SHOULD signal this state to the remote system if the transport protocol allows this.
 
-* Abort: The action the application may take on a Connection to indicate a Close, but with the additional indication that the transport system should not attempt to deliver any outstanding data.
+* Abort: The action the application takes on a Connection to indicate a Close and also indicate that the transport system SHOULD NOT attempt to deliver any outstanding data.
 
 ## Transport System Implementation Concepts
 
-The Transport System Implementation Concepts define the set of objects used internally to a system or library to provide the functionality required to provide a transport service across a network, as required by the abstract interface.
+The Transport System Implementation Concepts define the set of objects used internally to a system or library to provide the functionality needed to provide a transport service across a network, as required by the abstract interface.
```

Comment:
double "provide" here... maybe replace the second with "implement" ?

Reply:
Swapping the first "provide" for "implement", actually.